### ISSUE TITLE
[member-ordering] Support the "alphabetize" option from TSLint

### ISSUE THREAD

**Issue by octogonz** [typescript-eslint/typescript-eslint#780]
TSLint's [member-ordering](https://palantir.github.io/tslint/rules/member-ordering/) rule supports an "alphabetize" option, like this:

**Expected Result**
```ts
"member-ordering": [
  true,
  {
    "order": [
      "public-static-field",
      "public-instance-field",
      "public-constructor",
      "private-static-field",
      "private-instance-field",
      "private-constructor",
      "public-instance-method",
      "protected-instance-method",
      "private-instance-method"
    ],
    "alphabetize": true
  }
]
```

**Actual Result**

ESLint's [member-ordering](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/member-ordering.md) rule implements analogous orderings based on visibility and staticness -- but it lacks alphabetical ordering:

```ts
{
  default?: Array<MemberType> | never
  classes?: Array<MemberType> | never
  classExpressions?: Array<MemberType> | never

  interfaces?: ['field' | 'method' | 'constructor'] | never
  typeLiterals?: ['field' | 'method' | 'constructor'] | never
}
```

**Feature request:**  It would be nice if ESLint supported alphabetical ordering.

**Additional Info**

- We would probably not want alphabetization turned on by default. More likely, we'd instead use a comment directive to opt-in for classes that have lots of members.  
- Or, perhaps the rule itself could have a configurable threshold, that requires alphabetization only if the member count is greater than some number.
- It would be extremely valuable to have a fixer that could automatically sort members where this rule applies.  Otherwise, significant manual effort may be required to enable this for a legacy code base.

**Possible alternatives:**
- [eslint-plugin-typescript-sort-keys](https://www.npmjs.com/package/eslint-plugin-typescript-sort-keys) is a similar concept, but only applies to interfaces and enums (not classes)
- [eslint-plugin-sort-class-members](https://www.npmjs.com/package/eslint-plugin-sort-class-members) supports alphabetization, but its configuration is complicated, and it relies on [sort-class-members-codemod](https://github.com/pastelsky/sort-class-members-codemod) instead of a fixer.  Alphabetization seems like a simple enough feature that it shouldn't require overriding **member-ordering** with a plugin.
- We could also solve this by using **eslint-plugin-tslint**, but that's awkward.

What do you think?  Should **member-ordering** support alphabetization like TSLint?  If someone contributed a PR, would it be accepted?

**Comment by el** [comment]
I would like to be able to this too:
```
                "order": [
                    "public-static-field",
                    "protected-static-field",
                    "private-static-field",
                    "public-instance-field",
                    "protected-instance-field",
                    "private-instance-field",
                    "public-constructor",
                    "protected-constructor",
                    "private-constructor",
                    {
                        "name": "methods",
                        "kinds": [
                            "public-static-method",
                            "protected-static-method",
                            "private-static-method",
                            "public-instance-method",
                            "protected-instance-method",
                            "private-instance-method"
                        ],
                        "alphabetize": true
                    }
                ]

```

so we can enable `alphabetize` based on the `ruleset`

**Comment by octogonz** [comment]
Could you clarify the motivation behind that?  For example, why would you want to sort members of classes but not interfaces?

**Comment by el** [comment]
2 reasons:

1- `member-ordering` does not support readonly fields. We would like to be able to sort `readonly` fields first.
2- If a field depends on another, it should not be sorted solely based on its name:
```ts
class Foo {
  private static foo = "a";
  private static bar = `${Foo.foo} b`;
}
```

**Comment by octogonz** [comment]
For (1), your proposed configuration doesn't seem to consider readonly-or-not.  How would that be accounted for?

For (2), are you saying that `foo` should be sorted before `bar`?  But `f` comes after `b` in the alphabet.  Even if we were sorting by the assigned value instead of the member name, `$` comes before `a` according to the JavaScript `sort()` ordering.  Thus I'm not sure in what sense you're describing an `alphabetize` behavior.

**Comment by el** [comment]
1- Since `member-sort` does not differentiate `readonly`, I (wanted to) opted out of alphabetize for fields.
2- When you alphabetize without considering their use, you get an error stating the variable is used before its declaration. Therefor, I (wanted to) opted out of alphabetize for fields. 

So to recap, if `member-sort` had support for `readonly` fields and considered variable usage first, I would not need `alphabetize` for each ruleset inside `order`

**Comment by glen-84** [comment]
Would it be possible to alphabetize fields only, and not methods?

**Comment by grabofus** [comment]
I think as a start it'd be useful to just be able to use the rule as in TSLint, as for rationale, it's pretty much the same as ESLint's `sort-keys` (or any other sort related rule).
It allows you to have a consistent ordering in member groups, so you can navigate easier.

**Comment by shellscape** [comment]
@bradzacher looks like #263 was merged and this can be closed
